torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -156,7 +156,127 @@ def generate_ttir(kernel, kwargs):
     ttir_module = src.make_ir(options, context)
     if not ttir_module.verify():
         raise Exception("Verification for TTIR module has failed")
-    return str(ttir_module), ordered_tensor_names
+
+    return ttir_module, ordered_tensor_names
+
+
+def ttir_to_functions(ttir_module):
+    """
+    Walk the `ttir_module` bottom up to mine the `functions` from
+    the structured MLIR entities representing the Triton kernel
+    (mlir::Operation, mlir::Block, mlir::Region).
+    """
+    functions = {}
+
+    # block id --> op result (Intermediate) --> one or more ops
+    op_stack: Dict[int, Dict[Intermediate, List[Op]]] = defaultdict(
+        lambda: defaultdict(list)
```

Comment:
you shouldnt need the lambda?
```
defaultdict(defaultdict(list))
```
didn't work?

Reply:
I think, the inner `defaultdict(list)` would be an object, whereas the `default_factory` is expected to be a callable.